sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -116,6 +116,10 @@ object ColumnPruning extends Rule[LogicalPlan] {
     case a @ Aggregate(_, _, child) if (child.outputSet -- a.references).nonEmpty =>
       a.copy(child = Project(a.references.toSeq, child))
 
+    case p @ Project(pl, Aggregate(ge, ec, child))
+      if (ec.map(_.references).toSet -- p.references).nonEmpty =>
```

Comment:
indent 4.

Also I don't think this is doing what you want.  `ec.map(_.references).toSet` return a `Set[AttributeSet]`.

Instead you probably want `(a.outputSet -- p.references).nonEmpty` where `a` is bound to the `Aggregate`